pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -126,6 +126,7 @@ def _run_power_scaling(
 ) -> int:
     """Batch scaling mode where the size is doubled at each iteration until an OOM error is encountered."""
     for _ in range(max_trials):
+        trainer.reset_train_dataloader(model)  # FORCE RELOADING DATASET
```

Comment:
I think it should call the reset inside oom block down below because that's the place where we know that the current batch_size is not compatible with the current run and we need to decrease it and reload the dataloader.
```py
new_size, _ = _adjust_batch_size(trainer, batch_arg_name, factor=0.5, desc="failed")
trainer.reset_train_dataloader(model)
```